netshare/generate/denormalize_fields.py

```diff
@@ -52,26 +66,34 @@ def write_to_csv(
     session_key: List[np.ndarray],
     timeseries: List[np.ndarray],
     data_gen_flag: np.ndarray,
+    filename: str,
 ) -> None:
     """
     This function dumps the given data to the given directory as a csv format.
+    `data_gen_flag` is an indicator showing if the time series for this session
+    has ended in this time step.
     """
     os.makedirs(csv_folder, exist_ok=True)
-    csv_path = os.path.join(csv_folder, f"data_{random.random()}.csv")
+    csv_path = os.path.join(csv_folder, f"data_{filename}_{random.random()}.csv")
+    # change session key shape to #session * #attributes
+    session_key = np.concatenate(session_key, axis=1)
+    # change timeseries shape to #session * #time_steps * #features
+    timeseries = np.concatenate(timeseries, axis=2)
+
     with open(csv_path, "w") as f:
         writer = csv.writer(f)
         session_titles = _get_fields_names(session_key_fields)
         timeseries_titles = _get_fields_names(timeseries_fields)
         writer.writerow(session_titles + timeseries_titles)
 
-        for i in range(session_key[0].shape[0]):
-            session_data = [d[i] for d in session_key]
+        for i in range(data_gen_flag.shape[0]):
```

Comment:
```suggestion
        for data_gen, timeserie in zip(data_gen_flag, timeseries):
```
And then you can forget about the index `i`.
(Please read about it - I'm not sure that it will work with np.arrays, but if it doesn't - it helps for readability)